mmv1/templates/terraform/encoders/compute_instance_group_membership.go.erb

```diff
@@ -0,0 +1,23 @@
+<%# The license inside this block applies to this file.
+  # Copyright 2017 Google Inc.
+  # Licensed under the Apache License, Version 2.0 (the "License");
+  # you may not use this file except in compliance with the License.
+  # You may obtain a copy of the License at
+  #
+  #     http://www.apache.org/licenses/LICENSE-2.0
+  #
+  # Unless required by applicable law or agreed to in writing, software
+  # distributed under the License is distributed on an "AS IS" BASIS,
+  # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  # See the License for the specific language governing permissions and
+  # limitations under the License.
+-%>
+// Instance Group is a URL parameter only, so replace self-link/path with resource name only.
+if err := d.Set("instance_group", tpgresource.GetResourceNameFromSelfLink(d.Get("instance_group").(string))); err != nil {
+  return nil, fmt.Errorf("Error setting instance_group: %s", err)
+}
+obj["instance"] = flattenNestedComputeInstanceGroupMembershipInstance(d.Get("instance"), d, meta.(*transport_tpg.Config))
```

Comment:
Similar to the above, I don't believe we call flatteners in any of our existing encoders. Do you know if you need this line?

Reply:
Yes - it's needed because the `self_link` passed to `instance_group_membership` does not contains all the details and without this like the API returns us error as we're adding url to the instance that doesn't start with the googleapis.com

Reply:
Could you provide a little bit more detail here, I'm having trouble following exactly what you mean. I cloned this branch and removed the line here, and the tests still passed for me, so at first glance it does not seem necessary.

Looking a bit closer, it appears that the `obj["instance"]` that would be used by default does not have the full url (including googleapis.com), but I believe this is being done on purpose with the `expand*` function that is generated, and most APIs seem to support both formats. So I'm wondering if this API does not support both formats? If that's true, then I'm not sure why the test passed, and it's possible this should not be treated as a `ResourceRef`.

I'll note that the self_link/ResourceRef behavior is not an area of expertise for me, so if we get stuck, I can check with the team about the best option.

Reply:
Ok I've also removed it and it has passed. Most likely my bad with this.